inference-engine/src/mkldnn_plugin/nodes/extract_image_patches.cpp

```diff
@@ -1,248 +1,425 @@
-                IE_THROW() << errorPrefix << "must have 4D output tensor. Actual: " << layer->outData[0]->getTensorDesc().getDims().size();
-
-            if (inData->getLayout() != NCHW)
-                IE_THROW() << errorPrefix << "has unsupported layout: " << inData->getLayout();
-
-            const auto precision = inData->getTensorDesc().getPrecision();
-            if (_supported_precisions_sizes.find(precision.size()) == _supported_precisions_sizes.end())
-                IE_THROW() << errorPrefix << "has unsupported precision: " << precision.name();
-
-            auto ksizes = layer->GetParamAsUInts("sizes");
-            auto strides = layer->GetParamAsUInts("strides");
-            auto rates = layer->GetParamAsUInts("rates");
-            _auto_pad = layer->GetParamAsString("auto_pad");
-            if (!CaselessEq<std::string>()(_auto_pad, "valid")
-                    && !CaselessEq<std::string>()(_auto_pad, "same_upper")
-                    && !CaselessEq<std::string>()(_auto_pad, "same_lower"))
-                IE_THROW() <<  errorPrefix << "has unsupported auto_pad value: " << _auto_pad;
-            if (ksizes.size() != 2 || strides.size() != 2 || rates.size() != 2)
-                IE_THROW() << errorPrefix << "must have the following attributes with shape {2}: sizes, strides, rates.";
-
-            _ksizes.clear();
-            _strides.clear();
-            _rates.clear();
-            for (size_t i = 0; i < ksizes.size(); i++)
-                _ksizes.push_back((int64_t)ksizes[i]);
-            for (size_t i = 0; i < strides.size(); i++)
-                _strides.push_back((int64_t)strides[i]);
-            for (size_t i = 0; i < rates.size(); i++)
-                _rates.push_back((int64_t)rates[i]);
-
-            LayerConfig config;
-
-            DataConfig inConfig;
-            inConfig.desc = inData->getTensorDesc();
-            config.inConfs.push_back(inConfig);
-
-            DataConfig outConfig;
-            outConfig.desc = layer->outData[0]->getTensorDesc();
-            outConfig.desc.setPrecision(inConfig.desc.getPrecision());
-            outConfig.desc.setLayout(inConfig.desc.getLayout());
-            config.outConfs.push_back(outConfig);
-
-            config.dynBatchSupport = false;
-            confs.push_back(config);
-        } catch (InferenceEngine::Exception &ex) {
-            errorMsg = ex.what();
+using namespace dnnl::impl::cpu;
+using namespace dnnl::impl::cpu::x64;
+using namespace dnnl::impl::utils;
+using namespace dnnl::impl::cpu::x64;
+using namespace Xbyak;
+
+#define GET_OFF(field) offsetof(jit_eximpat_args, field)
+
+template <cpu_isa_t isa>
+struct jit_uni_eximpat_kernel_f32 : public jit_uni_eximpat_kernel, public jit_generator {
+    DECLARE_CPU_JIT_AUX_FUNCTIONS(jit_uni_eximpat_kernel_f32)
+
+    explicit jit_uni_eximpat_kernel_f32(jit_eximpat_params jpp) : jit_uni_eximpat_kernel(jpp), jit_generator() {}
```

Comment:
Looks like it is not just f32 kernel.

Reply:
You are right, I'll remove "_f32" postfix to avoid confusion.